### PR TITLE
add typeclass definition for cyclic groups 

### PR DESCRIPTION
**Pull request by santolucito**
along with helpful generation function. Not often you have a cyclic group, but when you do this can be nice to have.

### CODE REVIEW

**Comment by Taneb** [comment]
Thanks for the PR! This looks good, but is `ScopedTypeVariables` necessary? I'm trying to keep this as standard-Haskell2010 as possible

**Comment by santolucito** [comment]
I love to stick with Haskell2010 too, but tbh I'm not sure of another way to do it. The problem is that since the ```generated``` function has no parameters (its just a "constant"), it cannot know which ```Cyclic``` group instance it should use. In order to use the ```generator``` we somehow need to get type that we want to generate into the body of the function.

**Comment by Taneb** [comment]
The type of `generated`'s output decides which instance to use, so you can use a type annotation at the call site. ScopedTypeVariables don't actually add anything at all here

**Comment by santolucito** [comment]
Cool, dont know how i got it into my head that i did need it! Thanks for the close read! Fixed with 
26b9913

File: src/Data/Group.hs [hunk 1]
```diff
@@ -90,3 +90,14 @@ instance (Abelian a, Abelian b, Abelian c) => Abelian (a, b, c)
...
 
 instance (Abelian a, Abelian b, Abelian c, Abelian d, Abelian e) => Abelian (a, b, c, d, e)
+
+-- | A 'Group' G is 'Cyclic' if there exists an element x of G such that for all y in G, there exists an n, such that
+--
+-- @y = pow x n@
+class Group a => Cyclic a where
+  generator :: a
+
+generated :: (Cyclic a) => [a]
```
Comment (Taneb) [inline src/Data/Group.hs]: Minor style point, the parantheses around `Cyclic a` are redundant.
```suggestion
generated :: Cyclic a => [a]
```

**Pull request merged**